scene/animation/animation_mixer.cpp

```diff
@@ -2034,19 +2043,15 @@ void AnimationMixer::_call_object(ObjectID p_object_id, const StringName &p_meth
 	}
 }
 
-void AnimationMixer::make_animation_instance(const StringName &p_name, const PlaybackInfo p_playback_info) {
+void AnimationMixer::make_animation_instance(const StringName &p_name, const PlaybackInfo &p_playback_info) {
 	ERR_FAIL_COND(!has_animation(p_name));
-
-	AnimationData ad;
+	animation_instances.push_back(AnimationInstance());
+	AnimationInstance &ai = animation_instances[animation_instances.size() - 1];
+	AnimationData &ad = ai.animation_data;
```

Comment:
Wouldn't it be more efficient to create a constructor for AnimationInstance that takes an `AnimationData` and an `AnimationInstance` directly? This code will still run the constructor for `PlaybackInfo` unnecessarily.